### PR TITLE
New hydra slurm nodelist parse

### PR DESCRIPTION
**Pull request by gcongiu**
This series of patches provides better hydra/slurm integration for nodelist parsing

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

File: src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c [hunk 1]
```diff
@@ -9,9 +9,263 @@
...
+    regfree(&gmatch_new[0]);
+    regfree(&gmatch_old[1]);
+    regfree(&gmatch_new[1]);
+    regfree(&rmatch_old);
+    regfree(&rmatch_new);
+    regfree(&ematch_old);
+    regfree(&ematch_new);
+
+    /* free local nodelist */
+    MPL_free(string);
```
Comment (raffenet) [inline src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c]: This should use the old `HYDU_FREE` macro. The MPL updates aren't part of the 3.2.x series.

File: src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c [hunk 2]
```diff
@@ -35,12 +289,12 @@ static HYD_status group_to_nodes(char *str)
     *tmp = 0;   /* remove the closing ']' */
 
     /* Find the number of sets */
-    tnodes = HYDU_strdup(nodes);
+    tnodes = MPL_strdup(nodes);
```
Comment (raffenet) [inline src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c]: Should use `HYDU_strdup`.

File: src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c [hunk 3]
```diff
@@ -58,7 +312,7 @@ static HYD_status group_to_nodes(char *str)
         for (j = start; j <= end; j++) {
             char *node_str[HYD_NUM_TMP_STRINGS];
 
-            node_str[0] = HYDU_strdup(str);
+            node_str[0] = MPL_strdup(str);
```
Comment (raffenet) [inline src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c]: ditto

File: src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c [hunk 4]
```diff
@@ -35,12 +289,12 @@ static HYD_status group_to_nodes(char *str)
...
 
     /* Find the number of sets */
-    tnodes = HYDU_strdup(nodes);
+    tnodes = MPL_strdup(nodes);
     tmp = strtok(tnodes, ",");
     for (i = 1; tmp; i++)
         tmp = strtok(NULL, ",");
 
-    HYDU_MALLOC(set, char **, i * sizeof(char *), status);
+    HYDU_MALLOC_OR_JUMP(set, char **, i * sizeof(char *), status);
```
Comment (gcongiu) [inline src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c]: @raffenet should I also fix this?

Comment (raffenet) [reply src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c]: Yeah, it doesn't really belong in this series. Good catch.

**Comment by gcongiu** [comment]
test:jenkins/ch3/tcp

**Comment by gcongiu** [comment]
@raffenet should be good now. Sorry about the mess, I wrongly assumed the hydra code was the same in both branches.

**Comment by raffenet** [comment]
No problem. I'm sort of torn between keeping the history like it is here vs. squashing the fixup commits before merging this. I can't seem to find much in the way of best practice.

**Comment by gcongiu** [comment]
Squashing the fixup commits would mix two unrelated things. Isn't this unrecommended?

**Comment by raffenet** [comment]
I meant squashing these into the commits that added the features and not having fixups at all in the 3.2.x branch.

**Comment by raffenet** [comment]
Anyway it probably makes sense to just add all the commits as they are, so as to avoid any confusion between the 2 branches.

**Comment by raffenet** [comment]
All the signoffs are here already, so I'll just give this a quick sanity check before merge.
test:jenkins/ch3/tcp

**Comment by pavanbalaji** [comment]
Be careful with the signoffs.  People aren't being diligent enough to remove the signoffs when they edit a commit after it has been signed off on.

**Comment by raffenet** [comment]
Hmm, good point. These didn't apply cleanly. I can do a new review on the original 3 patches.

**Comment by raffenet** [comment]
Actually the first 3 did apply cleanly. It was the fixups that needed editing, which I can certify. Merging.

**Comment by raffenet** [comment]
Ugh, I forgot to update the fixup references in the commit messages :(.

**Pull request merged**